code/obj/machinery/manufacturer.dm

```diff
@@ -407,8 +407,9 @@ TYPEINFO(/obj/machinery/manufacturer)
 	proc/blueprints_as_list	(var/list/L, mob/user, var/static_elements = FALSE)
 		var/list/as_list = list()
 		for (var/datum/manufacture/M as anything in L)
-			if (isnull(M.category)) // fix for not displaying blueprints/manudrives
+			if (!(M.category in src.categories)) // fix for not displaying blueprints/manudrives
```

Comment:
```suggestion
			if (isnull(M.category) || !(M.category in src.categories)) // fix for not displaying blueprints/manudrives
```
Simple short-circuit for null case, clarifies that `null` is a possible value.

Reply:
fair